### PR TITLE
Make card have border on dark mode

### PR DESCRIPTION
**Pull request by ettiennelouw**
Added gray border-gray-600 on card for dark mode to outline of card and to title bottom border.

Before
image

After
image

### CODE REVIEW

**Comment by ph7jack** [comment]
@ettiennelouw I think that it can be a prop to enable borders in the card. 
@dansysanalyst @putera @trippo What's your option about it?

**Comment by putera** [comment]
> What's your option about it?

I think this is good for the dark mode. 👍🏽

**Comment by ettiennelouw** [comment]
I can make it a prop might make it more configurable. On light mode it has a shadow but then just give it a light shade of gray on light and the light shade on dark aswell but enable or disable with prop? I will make the change.

**Comment by ph7jack** [comment]
@ettiennelouw Will be good if you add an prop option to enable borders

**Comment by trippo** [comment]
I think is better to change the bg color of card

**Comment by ettiennelouw** [comment]
Made the border configurable but with defaults to match the rest of the components.

Default Light
Screenshot 2022-05-27 at 21 10 13

Default Dark
Screenshot 2022-05-27 at 21 10 19

Matches all components
Screenshot 2022-05-27 at 21 15 27

Custom
Pass to card  `<x-card title="Your title here" border="border-orange-600 dark:border-red-600"> ` to override with custom border styles.
Screenshot 2022-05-27 at 21 12 44

**Comment by ph7jack** [comment]
@ettiennelouw Great, thanks, but change the default card style without border in the dark mode. Many websites are using the card without borders in the dark mode, like the WireUI documentation, if enable the borders in the dark mode, the wireui docs will be strange.

**Comment by ettiennelouw** [comment]
@PH7-Jack instead of making a border property, I just moved the classes for the card, header and footer to the PHP class so that you can pass in the classes that you want for instance border, dark:border. Just had to do a little check on the header class as you had dark:border-0, if you can just check the logic in the file.

I think this might make it a bit more flexible and cater for all kinds of classes and styles someone wants to add to the card.

    <x-card title="Your title here" cardClasses="border border-gray-200 dark:border-secondary-600" headerClasses="dark:border-secondary-600">
    code sdf sdf sd fsd f sdf

         <x-slot name="footer">
             <div class="flex items-center justify-between">
                    <x-button label="Delete" flat negative />
                   <x-button label="Save" primary />
             </div>
        </x-slot>
    </x-card>

File: src/View/Components/Card.php [hunk 1]
```diff
@@ -22,28 +23,63 @@ class Card extends Component
...
+    }
+
+    public function setCardClasses($cardClasses)
+    {
+        return Str::of('w-full flex flex-col')->finish(' ')
+            ->append($this->shadow)->finish(' ')
+            ->append($this->rounded)->finish(' ')
+            ->append($this->color)->finish(' ')
+            ->append($cardClasses);
+    }
```
Comment (ph7jack) [inline src/View/Components/Card.php]: Missing split lines and the function return type

```suggestion
    public function setCardClasses(?string $cardClasses): string
    {
        return Str::of('w-full flex flex-col')
            ->append(" {$this->shadow}")
            ->append(" {$this->rounded}")
            ->append(" {$this->color}")
            ->append(" {$cardClasses}");
    }
```

File: src/View/Components/Card.php [hunk 2]
```diff
@@ -22,28 +23,63 @@ class Card extends Component
...
+    public function setHeaderClasses($headerClasses)
+    {
+        if (Str::contains($headerClasses, 'dark:border')) {
+            return Str::of('px-4 py-2.5 flex justify-between items-center border-b dark:border-0')->replace('dark:border-0', '')->finish(' ')
+            ->append($headerClasses);
+        } else {
+            return Str::of('px-4 py-2.5 flex justify-between items-center border-b dark:border-0')->finish(' ')
+            ->append($headerClasses);
+        }
+    }
```
Comment (ph7jack) [inline src/View/Components/Card.php]:
```suggestion
    public function setHeaderClasses(?string $headerClasses): string
    {
        if (Str::contains($headerClasses, 'dark:border')) {
            return Str::of('px-4 py-2.5 flex justify-between items-center border-b dark:border-0')
                ->replace('dark:border-0', '')
                ->append(" {$headerClasses}");
        }
        return Str::of('px-4 py-2.5 flex justify-between items-center border-b dark:border-0')
            ->append(" {$headerClasses}");
    }
```

File: src/View/Components/Card.php [hunk 3]
```diff
@@ -22,28 +23,63 @@ class Card extends Component
...
+
+    public function setFooterClasses($footerClasses)
+    {
+        return Str::of('px-4 py-4 sm:px-6 bg-secondary-50 rounded-t-none border-t dark:bg-secondary-800 dark:border-secondary-600')->finish(' ')
+            ->append($this->rounded)->finish(' ')
+            ->append($footerClasses);
```
Comment (ph7jack) [inline src/View/Components/Card.php]:
```suggestion
    public function setFooterClasses(?string $footerClasses): string
    {
        return Str::of('px-4 py-4 sm:px-6 bg-secondary-50 rounded-t-none border-t dark:bg-secondary-800 dark:border-secondary-600')
            ->append(" {$this->rounded}")
            ->append(" {$footerClasses}");
```

**Review by ph7jack — DISMISSED** [review]
Nice job, thank you. Could you apply these suggestions?

**Comment by ettiennelouw** [comment]
Awsome all comitted.

**Review by ph7jack — APPROVED** [review]
Thanks for contributing

**Pull request merged**